### PR TITLE
Add TestTombstoneResync

### PR DESCRIPTION
**Pull request by mohammed-madi**
Describe your PR here...
- Added test that runs query based resync on a bucket full of tombstones

## Pre-review checklist
- [ ] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [ ] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [ ] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## Dependencies (if applicable)
- [ ] Link upstream PRs
- [ ] Update Go module dependencies when merged

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1943/

### CODE REVIEW

File: rest/changestest/changes_api_test.go [hunk 1]
```diff
@@ -3856,6 +3856,62 @@ func TestCacheCompactDuringChangesWait(t *testing.T) {
 	longpollWg.Wait()
 }
 
+func TestTombstoneResync(t *testing.T) {
```
Comment (bbrks) [inline rest/changestest/changes_api_test.go]:
```suggestion
func TestResyncAllTombstonest *testing.T) {
```

Maybe more accurate?

File: rest/changestest/changes_api_test.go [hunk 2]
```diff
@@ -3856,6 +3856,62 @@ func TestCacheCompactDuringChangesWait(t *testing.T) {
...
+			DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+				QueryPaginationLimit: base.IntPtr(5),
+				Unsupported:          &db.UnsupportedOptions{UseQueryBasedResyncManager: true}}}}, 2)
+	} else {
+		rt = rest.NewRestTester(t, &rest.RestTesterConfig{
+			DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+				QueryPaginationLimit: base.IntPtr(5),
+				Unsupported:          &db.UnsupportedOptions{UseQueryBasedResyncManager: true}}},
+		})
+	}
```
Comment (bbrks) [inline rest/changestest/changes_api_test.go]: This collection-specific code isn't required as both configs are the same and you don't need to know the number of collections in use for assertions like the test you got this setup code from.

```suggestion
	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
		DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
			QueryPaginationLimit: base.IntPtr(5),
			Unsupported:          &db.UnsupportedOptions{UseQueryBasedResyncManager: true}}},
	})
```

**Review by bbrks — CHANGES_REQUESTED** [review]

File: rest/changestest/changes_api_test.go (hunk 2 above)
Comment (mohammed-madi) [reply rest/changestest/changes_api_test.go]: https://github.com/couchbase/sync_gateway/pull/6340/commits/b3111c121362d3ddd55aaa0acb9344f6c20178da

File: rest/changestest/changes_api_test.go (hunk 1 above)
Comment (mohammed-madi) [reply rest/changestest/changes_api_test.go]: https://github.com/couchbase/sync_gateway/pull/6340/commits/b3111c121362d3ddd55aaa0acb9344f6c20178da

**Review by bbrks — APPROVED** [review]

**Pull request merged**